compiler/cpp/src/thrift/generate/t_csharp_generator.cc

```diff
@@ -1254,10 +1254,10 @@ void t_csharp_generator::generate_csharp_union_definition(std::ostream& out, t_s
   indent_up();
 
   indent(out) << "public abstract void Write(TProtocol protocol);" << endl;
-  indent(out) << "public readonly bool Isset;" << endl;
-  indent(out) << "public abstract object Data { get; }" << endl;
+  indent(out) << "public readonly int Isset;" << endl;
+  indent(out) << "protected abstract object Data { get; }" << endl;
```

Comment:
This looks like it would be a breaking change for folks who grab the data directly?  If so, it should be documented in the lib/csharp/README.md and in the top level CHANGES.md file.

Reply:
I thought a while about this as well. I think I change it back to public, so the user has all the options (and can shoot himself in the foot in the way he likes most 🤣 ).

Reply:
> Should the union option be the default (on the compiler)? 

Now *that* would certainly be a breaking change.

Reply:
Well perhaps in netstd it should be the default (if it's better).